File: storage/data_transpiled/at_coder/agc015/D/4201561.java
public static int bitLen ( ) {
  int a = Integer . parseInt ( input . nextLine ( ) ) ;
  int b = Integer . parseInt ( input . nextLine ( ) ) ;
  if ( a == b ) {
    System . out . println ( 1 ) ;
    exit ( ) ;
  }
  /* bit length */
  {
    return a > 0 ? 1 + bitLen ( a >> 1 ) : 0 ;
  }
  int t = a ^ b ;
  int n = bitLen ( t ) ;
  int T = 1 << ( n - 1 ) ;
  a = a & ( T * 2 - 1 ) ;
  b = b & ( T * 2 - 1 ) ;
  int sb = b & ( T - 1 ) ;
  int bMax = ( 2 * bitLen ( sb ) - 1 ) | T ;
  if ( ( T + a ) > bMax ) {
    System . out . println ( ( bMax - a + 1 ) + ( 2 * T - 1 - ( T + a ) + 1 ) ) ;
  }
  else {
    System . out . println ( 2 * T - 1 - a + 1 ) ;
  }
  return t ;
}

File: data/data_original/at_coder/agc015/D/4201561.py
a = int(input())
b = int(input())

if a == b:
    print(1)
    exit()

# ???x?????????len(bin(x)) - 2????x?0???1????????????????
# ????????????????????????
def bitlen(x):
    return 1 + bitlen(x >> 1) if x else 0

t = a ^ b
n = bitlen(t)

T = 1 << (n - 1)

a = a & (T * 2 - 1)
b = b & (T * 2 - 1)

sb = b & (T - 1)
b_max = (2 ** bitlen(sb) - 1) | T

if (T + a) > b_max:
    print((b_max - a + 1) + (2 * T - 1 - (T + a) + 1))
else:
    print(2 * T - 1 - a + 1)